Framework/API/inc/MantidAPI/GeometryInfoFactory.h

```diff
@@ -0,0 +1,81 @@
+#ifndef MANTID_API_GEOMETRYINFOFACTORY_H_
+#define MANTID_API_GEOMETRYINFOFACTORY_H_
+
+#include <boost/shared_ptr.hpp>
+
+#include "MantidKernel/V3D.h"
+#include "MantidAPI/DllConfig.h"
+
+namespace Mantid {
+
+namespace Geometry {
+class Instrument;
+class IComponent;
+}
+
+namespace API {
+class MatrixWorkspace;
+class GeometryInfo;
+
+/** Factory for GeometryInfo, see there for detailed information.
+
+  @author Simon Heybrock, ESS
+
+  Copyright &copy; 2016 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+  National Laboratory & European Spallation Source
+
+  This file is part of Mantid.
+
+  Mantid is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  Mantid is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  File change history is stored at: <https://github.com/mantidproject/mantid>
+  Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_API_DLL GeometryInfoFactory {
```

Comment:
`GeometryInfoFactory` makes me think the sole responsibility of this is creational, i.e. is to create `GeometryInfo`, I think the relationship is actually more complex than this.